app/assets/javascripts/cascade/new-home.js

```diff
@@ -1,1139 +1,1189 @@
-			];
-			var html = jQuery('html'); // it would make more sense to apply this to body, but IE7 won't have that
-			html.data('scroll-position', scrollPosition);
-			html.data('previous-overflow', html.css('overflow'));
-			html.css('overflow', 'hidden');
-			window.scrollTo(scrollPosition[0], scrollPosition[1]);
-			this.isScrollLocked = true;
-		}, // end lockScroll
-
-		unlockScroll : function() {
-			if (!this.isScrollLocked) return false;
-
-			var html = jQuery('html');
-			var scrollPosition = html.data('scroll-position');
-			html.css('overflow', html.data('previous-overflow'));
-			window.scrollTo(scrollPosition[0], scrollPosition[1]);
-		},
-
-		hasImageExtension : function(url) {
-			return(url.match(/\.(jpeg|jpg|gif|png)$/) != null);
-		}
-	} // end heroQuickView
-	heroQuickView.initialize();
-
-
-})(jQuery);
-
-// Define Lazybind
-(function($){
-	$.fn.lazybind = function(event, fn, timeout, abort){
-		var timer = null;
-		$(this).bind(event, function(e){
-			var ev = e;
-			timer = setTimeout(function(){
-				fn(ev);
-			}, timeout);
-		});
-		if(abort == undefined){
-			return;
-		}
-		$(this).bind(abort, function(){
-			if(timer != null){
-				clearTimeout(timer);
-			}
-		});
-	};
-})(jQuery);
-
-// Object.keys pollyfill for older IE
-if (!Object.keys) {
-  Object.keys = (function () {
-	var hasOwnProperty = Object.prototype.hasOwnProperty,
-		hasDontEnumBug = !({toString: null}).propertyIsEnumerable('toString'),
-		dontEnums = [
-		  'toString',
-		  'toLocaleString',
-		  'valueOf',
-		  'hasOwnProperty',
-		  'isPrototypeOf',
-		  'propertyIsEnumerable',
-		  'constructor'
-		],
-		dontEnumsLength = dontEnums.length;
-
-	return function (obj) {
-	  if (typeof obj !== 'object' && typeof obj !== 'function' || obj === null) throw new TypeError('Object.keys called on non-object');
-
-	  var result = [];
-
-	  for (var prop in obj) {
-		if (hasOwnProperty.call(obj, prop)) result.push(prop);
-	  }
-
-	  if (hasDontEnumBug) {
-		for (var i=0; i < dontEnumsLength; i++) {
-		  if (hasOwnProperty.call(obj, dontEnums[i])) result.push(dontEnums[i]);
-		}
-	  }
-	  return result;
-	}
-  })()
-};
-
-// Array.forEach pollyfill for older IE
-if (!Array.prototype.forEach) {
-  Array.prototype.forEach = function(fun /*, thisp*/)
-  {
-	var len = this.length;
-	if (typeof fun != "function")
-	  throw new TypeError();
-	var thisp = arguments[1];
-	for (var i = 0; i < len; i++)
-	{
-	  if (i in this)
-		fun.call(thisp, this[i], i, this);
-	}
-  };
-};
\ No newline at end of file
+(function ($) {
+    $(document).ready(function() {
+        var onHomePage = $('.homepage').length > 0;
+
+        if ( onHomePage ) {
+            cu_window_manager.initialize();
+            cu_hero_area.initialize();
+            cu_stories_area.initialize();
+            cu_admission_area.initialize();
+            smc_cta_tracker.initialize();
+            heroModalViewer.initialize();
+        }
+    });
```

Comment:
Doing this a little more sensibly now. All the module code below used to be in the document ready callback, too.